src/NUnitTestAdapter/NUnitTestFilterBuilder.cs

```diff
@@ -29,15 +29,22 @@ public TestFilter ConvertTfsFilterToNUnitFilter(TfsTestFilter tfsFilter, List<Te
             return MakeTestFilter(testCases);
         }
 
-        public TestFilter MakeTestFilter(IEnumerable<TestCase> testCases)
+        public TestFilter MakeTestFilter(IEnumerable<TestCase> testCases = null, string where = null)
         {
-            if (testCases.Count() == 0)
-                return NoTestsFound;
-
             ITestFilterBuilder filterBuilder = _filterService.GetTestFilterBuilder();
 
-            foreach (TestCase testCase in testCases)
-                filterBuilder.AddTest(testCase.FullyQualifiedName);
+            if(!string.IsNullOrEmpty(where))
+            {
+                filterBuilder.SelectWhere(where);
+            }
+
+            if(testCases is null == false)
```

Comment:
Is there any reason why  if (testcases != null) is not used ?

Reply:
Using `is` avoids any potential equality overload shenanigans. Though perhaps `testCases?.Any()` would be better.

Reply:
Yes, testcases?.Any sounds good :-)

Reply:
This situation was eliminated due to the above refactor.